src/Core/Types.Filters/FilterInputTypeDescriptor.cs

```diff
@@ -283,8 +323,59 @@ private bool IsComparable(Type type)
                 nameof(property));
         }
 
+
+
+        public IArrayFilterFieldDescriptor<TObject> ListFilter<TObject, TListType>(
+            Expression<Func<T, TListType>> property)
+        {
+            if (property.ExtractMember() is PropertyInfo p)
+            {
+                var field = new ArrayFilterFieldDescriptor<TObject>(Context, p);
+                Fields.Add(field);
+                return field;
+            }
+
+            throw new ArgumentException(
+                FilterResources.FilterInputTypeDescriptor_OnlyProperties,
+                nameof(property));
+        }
+
+        public IArrayFilterFieldDescriptor<TObject> List<TObject>(
+            Expression<Func<T, IEnumerable<TObject>>> property)
+            where TObject : class
+        {
+            return ListFilter<TObject, IEnumerable<TObject>>(property);
+        }
+
+
+        public IArrayFilterFieldDescriptor<ISingleFilter<string>> List(Expression<Func<T, IEnumerable<string>>> property)
+        {
+            return ListFilter<ISingleFilter<string>, IEnumerable<string>>(property);
+        }
+
+        public IArrayFilterFieldDescriptor<ISingleFilter<bool>> List(Expression<Func<T, IEnumerable<bool>>> property)
+        {
+            return ListFilter<ISingleFilter<bool>, IEnumerable<bool>>(property);
+        }
+
+
+        public IArrayFilterFieldDescriptor<ISingleFilter<TStruct>> List<TStruct>(
+            Expression<Func<T, IEnumerable<TStruct>>> property,
+            IFilterInputTypeDescriptor<T>.RequireStruct<TStruct> ignore = null) where TStruct : struct
```

Comment:
@michaelstaib  You will probably not like this. This was the only way I found to capture the generics correctly...

Reply:
I still think it is a bit hacky. Have we enough tests around this?

Reply:
I is hacky indeed.. I played around a while and didn't find a better way. This is only used for capturing generics. So its only invoked by the user in the descriptor. 

There are plenty tests.  If you find a better way I'm up changing it. If you want to play around a little you can just do changes on this and see if all the test run through